safekeeper/src/wal_service.rs

```diff
@@ -68,28 +69,30 @@ fn handle_socket(
         .build()?;
     let local = tokio::task::LocalSet::new();
 
-    let read_metrics = PG_IO_BYTES.with_label_values(&["read"]);
-    let write_metrics = PG_IO_BYTES.with_label_values(&["write"]);
-
     socket.set_nodelay(true)?;
     let peer_addr = socket.peer_addr()?;
 
-    // TODO: measure cross-az traffic
+    let traffic_metrics = TrafficMetrics::new();
+    if let Some(current_az) = conf.availability_zone.as_deref() {
+        traffic_metrics.set_sk_az(current_az);
+    }
+
     let socket = MeasuredStream::new(
         socket,
         |cnt| {
-            read_metrics.inc_by(cnt as u64);
+            traffic_metrics.observe_read(cnt);
         },
         |cnt| {
-            write_metrics.inc_by(cnt as u64);
+            traffic_metrics.observe_write(cnt);
         },
     );
 
     let auth_type = match conf.auth {
         None => AuthType::Trust,
         Some(_) => AuthType::NeonJWT,
     };
-    let mut conn_handler = SafekeeperPostgresHandler::new(conf, conn_id);
+    let mut conn_handler =
+        SafekeeperPostgresHandler::new(conf, conn_id, Some(traffic_metrics.clone()));
```

Comment:
Shouldn't there be clippy warning about extra clone here or ... cannot see any other use for the value in this scope?

Reply:
It's borrowed in the closures:
```
error[E0505]: cannot move out of `traffic_metrics` because it is borrowed
  --> safekeeper/src/wal_service.rs:95:60
   |
82 |         |cnt| {
   |         ----- borrow of `traffic_metrics` occurs here
83 |             traffic_metrics.observe_read(cnt);
   |             --------------- borrow occurs due to use in closure
...
95 |         SafekeeperPostgresHandler::new(conf, conn_id, Some(traffic_metrics));
   |                                                            ^^^^^^^^^^^^^^^ move out of `traffic_metrics` occurs here
96 |     let pgbackend = PostgresBackend::new_from_io(socket, peer_addr, auth_type, None)?;
   |                                                  ------ borrow later used here

For more information about this error, try `rustc --explain E0505`.
error: could not compile `safekeeper` due to previous error
```